File: practices/practice3/gcd.cpp
#include<iostream>
using namespace std;

int gcd(int a, int b) {
    if(b == 0)
        return a;
    else
        return gcd(b, a%b);
}

int main() {
    cout << gcd(10, 5);
    return 0;
}

File: practices/practice3/GCD.scala
object GCD {
  def main(args: Array[String]): Unit = {
    println(gcd(12, 4))
  }
  def gcd(a: Int, b: Int): Int =
    if (a%b==0)
      b
    else {
      var r: Int = a%b
      gcd(b, a%b)
    }
}
